bind_slurm.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# Specify system-specific bindings to use here.
+
+CONFIG=nesh.conf
+
+
+
+# Load bindings from config.
+
+source ${CONFIG}
+
+echo Will use: ${CONFIG}
+
+# Check if singularity is available.
+
+singularity --version || { echo Singularity not found... exiting.; exit; }
+
+# Get Singularity image path.
+
+ORIGINAL_SINGULARITY_COMMAND="$@" # Preserve original argument
+
+echo Will execute: ${ORIGINAL_SINGULARITY_COMMAND}
+
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    *.sif) IMAGE="$1" ;;
+    *) ;;
+  esac
+  shift
+done
+
+echo Enable host SLURM user for: ${IMAGE}
+
+# Prepare system-specific SLURM user binding.
+
+TEMPDIR=$(mktemp -d $(pwd)/tmp.XXXXXXXX)
+echo Temporary directory: ${TEMPDIR}
+trap "echo Deleting: ${TEMPDIR}; rm -rf ${TEMPDIR}" 0
+
+singularity exec ${IMAGE} cp -p /etc/passwd ${TEMPDIR}/etc_passwd
+singularity exec ${IMAGE} cp -p /etc/group ${TEMPDIR}/etc_group
+cat /etc/passwd | grep slurm >> ${TEMPDIR}/etc_passwd
+cat /etc/group | grep slurm >> ${TEMPDIR}/etc_group
+
+#getent passwd >> ${TEMPDIR}/etc_passwd
+#getent group >> ${TEMPDIR}/etc_group
+
+RTEMPDIR=$(basename $TEMPDIR) # get relative temp dir path
+
+MERGED_PASSWD_GROUP="\
+${RTEMPDIR}/etc_passwd:/etc/passwd,\
+${RTEMPDIR}/etc_group:/etc/group\
+"
+
```

Comment:
```suggestion
echo Enable host SLURM user for: ${IMAGE}

# Prepare binding of system-specific SLURM.
#
# We'll filter system users and if there is a slurm user, we'll add it to the 
# /etc/passwd and /etc/group files used in the container.
# 
# This is done very similar to how singularity enables the human user inside the container:
# First, briefly start the container to get /etc/passwd and /etc/groups from with, then
# append users and groups to the files and bind the files when finally starting the container.

# create temporary directory that'll be deleted automatically once the script exists successfully
TEMPDIR=$(mktemp -d $(pwd)/tmp.XXXXXXXX)
echo Temporary directory: ${TEMPDIR}
trap "echo Deleting: ${TEMPDIR}; rm -rf ${TEMPDIR}" 0

# Extract user and group info from container and (mayby) append the slurm user
singularity exec ${IMAGE} cp -p /etc/passwd ${TEMPDIR}/etc_passwd
singularity exec ${IMAGE} cp -p /etc/group ${TEMPDIR}/etc_group
grep slurm /etc/passwd >> ${TEMPDIR}/etc_passwd
grep slurm /etc/group >> ${TEMPDIR}/etc_group

# Note that if system users are not managed in /etc/..., we could use getent:
# getent passwd | grep slurm >> ${TEMPDIR}/etc_passwd
# getent group | grep slurm >> ${TEMPDIR}/etc_group

# Create bind paths
RTEMPDIR=$(basename $TEMPDIR) # get relative temp dir path

MERGED_PASSWD_GROUP="\
${RTEMPDIR}/etc_passwd:/etc/passwd,\
${RTEMPDIR}/etc_group:/etc/group\
"

```

Reply:
I will fix some typos, but commit this for now! :)